cabal-install/src/Distribution/Client/CmdRepl.hs

```diff
@@ -324,77 +348,9 @@ data OriginalComponentInfo = OriginalComponentInfo
-data ReplType = ProjectRepl
-              | GlobalRepl FilePath -- ^ The 'FilePath' argument is path to a GHCi
-                                    --   script responsible for changing to the
-                                    --   correct directory. Only works on GHC geq
-                                    --   7.6, though. 🙁
-              deriving (Show, Eq)
-
-withProject :: ProjectConfig -> Verbosity -> [String]
-            -> IO (ProjectBaseContext, [TargetSelector], IO (), ReplType)
-withProject cliConfig verbosity targetStrings = do
-  baseCtx <- establishProjectBaseContext verbosity cliConfig OtherCommand
-
-  targetSelectors <- either (reportTargetSelectorProblems verbosity) return
-                 =<< readTargetSelectors (localPackages baseCtx) (Just LibKind) targetStrings
-
-  return (baseCtx, targetSelectors, return (), ProjectRepl)
-
-withoutProject :: ProjectConfig -> Verbosity -> [String]
-               -> IO (ProjectBaseContext, [TargetSelector], IO (), ReplType)
-withoutProject config verbosity extraArgs = do
-  unless (null extraArgs) $
-    die' verbosity $ "'repl' doesn't take any extra arguments when outside a project: " ++ unwords extraArgs
-
-  globalTmp <- getTemporaryDirectory
-  tempDir <- createTempDirectory globalTmp "cabal-repl."
-
-  -- We need to create a dummy package that lives in our dummy project.
-  let
-    sourcePackage = SourcePackage
-      { srcpkgPackageId     = pkgId
-      , srcpkgDescription   = genericPackageDescription
-      , srcpkgSource        = LocalUnpackedPackage tempDir
-      , srcpkgDescrOverride = Nothing
-      }
-    genericPackageDescription = emptyGenericPackageDescription
-      & L.packageDescription .~ packageDescription
-      & L.condLibrary        .~ Just (CondNode library [baseDep] [])
-    packageDescription = emptyPackageDescription
-      { package = pkgId
-      , specVersion = CabalSpecV2_2
-      , licenseRaw = Left SPDX.NONE
-      }
-    library = emptyLibrary { libBuildInfo = buildInfo }
-    buildInfo = emptyBuildInfo
-      { targetBuildDepends = [baseDep]
-      , defaultLanguage = Just Haskell2010
-      }
-    baseDep = Dependency "base" anyVersion mainLibSet
-    pkgId = fakePackageId
-
-  writeGenericPackageDescription (tempDir </> "fake-package.cabal") genericPackageDescription
-
-  let ghciScriptPath = tempDir </> "setcwd.ghci"
-  cwd <- getCurrentDirectory
-  writeFile ghciScriptPath (":cd " ++ cwd)
-
-  distDirLayout <- establishDummyDistDirLayout verbosity config tempDir
-  baseCtx <-
-    establishDummyProjectBaseContext
-      verbosity
-      config
-      distDirLayout
-      [SpecificSourcePackage sourcePackage]
-      OtherCommand
-
-  let
-    targetSelectors = [TargetPackage TargetExplicitNamed [pkgId] Nothing]
-    finalizer = handleDoesNotExist () (removeDirectoryRecursive tempDir)
-
-  return (baseCtx, targetSelectors, finalizer, GlobalRepl ghciScriptPath)
+-- Tracks what type of GHCi instance we're creating.
```

Comment:
```suggestion
-- | Tracks what type of GHCi instance we're creating.
```

Haddock ftw! Because HLS can show you this documentation and believe me, you want this ;D 

Additionally, maybe mention why or how we handle this differently? (Just bonus though, I am happy with haddock comment)